### ISSUE TITLE
With httplib2 0.14.0 many tests fail

### ISSUE THREAD

**Issue by scarabeusiv** [cdent/wsgi-intercept#57]
```
________________________________________________________________________________________________ test_bogus_domain ________________________________________________________________________________________________
wsgi_intercept/tests/test_httplib2.py:47: in test_bogus_domain
    'httplib2_intercept.HTTP_WSGIInterceptorWithTimeout('
E   TypeError: 'httplib2_intercept.HTTP_WSGIInterceptorWithTimeout("_nonexistant_domain_").connect()' object (type: <class 'str'>) must be callable
___________________________________________________________________________________________________ test_https ____________________________________________________________________________________________________
wsgi_intercept/tests/test_httplib2.py:66: in test_https
    'https://some_hopefully_nonexistant_domain:443/')
.tox/py37/lib/python3.7/site-packages/httplib2/__init__.py:1794: in request
    tls_minimum_version=self.tls_minimum_version,
E   TypeError: __init__() got an unexpected keyword argument 'tls_maximum_version'
_____________________________________________________________________________________________ test_https_default_port _____________________________________________________________________________________________
wsgi_intercept/tests/test_httplib2.py:74: in test_https_default_port
    'https://some_hopefully_nonexistant_domain/')
.tox/py37/lib/python3.7/site-packages/httplib2/__init__.py:1794: in request
    tls_minimum_version=self.tls_minimum_version,
E   TypeError: __init__() got an unexpected keyword argument 'tls_maximum_version'
______________________________________________________________________________________ test_httplib2_interceptor_https_host _______________________________________________________________________________________
wsgi_intercept/tests/test_interceptor.py:125: in test_httplib2_interceptor_https_host
    response, content = http.request(url)
.tox/py37/lib/python3.7/site-packages/httplib2/__init__.py:1794: in request
    tls_minimum_version=self.tls_minimum_version,
E   TypeError: __init__() got an unexpected keyword argument 'tls_maximum_version'
________________________________________________________________________________________________ test_bogus_domain ________________________________________________________________________________________________
wsgi_intercept/tests/test_requests.py:41: in test_bogus_domain
    'requests.get("http://_nonexistant_domain_")')
E   TypeError: 'requests.get("http://_nonexistant_domain_")' object (type: <class 'str'>) must be callable
________________________________________________________________________________________________ test_bogus_domain ________________________________________________________________________________________________
wsgi_intercept/tests/test_urllib3.py:43: in test_bogus_domain
    'http.request("GET", "http://_nonexistant_domain_", '
E   TypeError: 'http.request("GET", "http://_nonexistant_domain_", retries=False)' object (type: <class 'str'>) must be callable
______________________________________________________________________________________________ test_https_in_environ ______________________________________________________________________________________________
wsgi_intercept/tests/test_wsgi_compliance.py:37: in test_https_in_environ
    'https://some_hopefully_nonexistant_domain/', 'GET')
.tox/py37/lib/python3.7/site-packages/httplib2/__init__.py:1794: in request
    tls_minimum_version=self.tls_minimum_version,
E   TypeError: __init__() got an unexpected keyword argument 'tls_maximum_version'

```

**Comment by scarabeusiv** [comment]
@temoto this might be a bug in httplib2 and I can trace it to the 0.13.0 too, 0.12.3 was passing all the tests here.

**Comment by scarabeusiv** [comment]
By here I mean that if you look on the travis run of the last commit it is passing because it used the old httplib2.

**Comment by cdent** [comment]
I suspect the main issue here is that the interface in httplib2 has changed slightly and the rather tricky monkey-patching that wsgi-intercept is doing no longer works.

I'll investigate this quickly and if I don't see a quick path to fixing it I'll do a release that pins to an earlier version of httplib2 and do the harder fix later.

**Comment by temoto** [comment]
Culprit: https://github.com/cdent/wsgi-intercept/blob/master/wsgi_intercept/httplib2_intercept.py#L33

```python
    def __init__(self, host, port=None, strict=None, timeout=None,
            proxy_info=None, ca_certs=None, source_address=None,
            disable_ssl_certificate_validation=False,
            ssl_version=None):
```

Solution: accept **kwargs.

**Comment by cdent** [comment]
> Solution: accept **kwargs.

Yes, that's where I'm at.

The original authors didn't use kwargs so I maintained the pattern when I took over maintenance, but it no longer seems worth the effort.

**Comment by cdent** [comment]
see #58

**Comment by scarabeusiv** [comment]
I can confirm #58 fixes the issue and the tests now pass.

**Comment by cdent** [comment]
I've released a 1.9.0 that has the fix: https://pypi.org/project/wsgi-intercept/

**Comment by cdent** [comment]
This was fixed way back when by #58 so should have been closed then.
